packages/dashboard/src/components/robots/robot-alert.tsx

```diff
@@ -0,0 +1,158 @@
+import React from 'react';
+import { RobotState, Status2 } from 'api-client';
+import { RmfAppContext } from '../rmf-app';
+import { AlertProps, AlertToDisplay, RobotWithTask } from '../alert-store';
+import { AlertDialog, AlertContent } from '../alert-dialog-component';
+import { base } from 'react-components';
+
+const statusToAlert = (robot: RobotState) => {
+  switch (robot.status) {
+    case Status2.Error:
+    case Status2.Offline:
+      return true;
+    default:
+      return false;
+  }
+};
+
+const setRobotDialogColor = (robotStatus: Status2 | undefined) => {
+  if (!robotStatus) {
+    return base.palette.background.default;
+  }
+
+  switch (robotStatus) {
+    case Status2.Error:
+      return base.palette.error.main;
+
+    case Status2.Offline:
+      return base.palette.warning.main;
+
+    default:
+      return base.palette.background.default;
+  }
+};
+
+const showMessage = (robot: RobotState) => {
+  switch (robot.status) {
+    case Status2.Error:
+      if (robot.issues) {
+        robot.issues.map((i) => {
+          if (!i.detail) {
+            return 'No Detail';
+          }
+          if (Array.isArray(i.detail)) {
+            i.detail.map(
+              (d) => ` * ${d} 
+                `,
+            );
+          }
+          return i.detail;
+        });
+      }
```

Comment:
@aaronchongth I'm sorry I didn't fully understand you. 
What do you mean when you say that it will only parse the first issue it encounters in the robot issues list?

This should go through the entire list of issues and also if the details field is an array go through each detail and display it. 

In case it is not an array, each detail field of each issue of issues will be displayed.

Reply:
Apologies I misunderstood what the issue was. Thanks for explaining!